igcollect/switch.py

```diff
@@ -328,6 +355,23 @@ def ports_stats(prefix, snmp, ports, model):
                 print(template.format(port_name, counter, data))
 
 
+def dom_stats(prefix, snmp, ports, model):
+    """
+    Print graphite-compatible Digital Optical Monitoring stats for each port
+    of the switch
+    """
+
+    for metric, oid in DOM_OIDS[model].items():
+        # SNMP is slow, so we want need to get current time for each counter.
+        template = prefix + '.ports.{}.{} {} ' + str(int(time()))
+        table = get_snmp_table(snmp, oid)
+        for port_idx, port_name in ports.items():
+            if port_idx in table:
+                data = table[port_idx]
+                if data and data != '':
+                    print(template.format(port_name, metric, data))
```

Comment:
Suggestion to @Woellchen 's comment:
```suggestion
        for port_idx, port_name in ports.items():
            if port_idx not in table:
                continue
            data = table[port_idx]
            if not data:
                continue
            print(template.format(port_name, metric, data))
```

Reply:
Thank you. I committed the suggestion.